src/aggregated_range_proof/tests.rs

```diff
@@ -0,0 +1,106 @@
+use curve25519_dalek::scalar::Scalar;
+use proof_transcript::ProofTranscript;
+use rand::Rng;
+use std::iter;
+
+use super::dealer::*;
+use super::messages::*;
+use super::party::*;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand::OsRng;
+
+    fn create_multi<R: Rng>(rng: &mut R, values: Vec<u64>, n: usize) -> Proof {
+        use generators::{Generators, PedersenGenerators};
+
+        let m = values.len();
+        let generators = Generators::new(PedersenGenerators::default(), n, m);
+        let mut transcript = ProofTranscript::new(b"AggregatedRangeProofTest");
+
+        let parties: Vec<_> = values
+            .iter()
+            .map(|&v| {
+                let v_blinding = Scalar::random(rng);
+                Party::new(v, v_blinding, n, &generators)
+            })
+            .collect();
+
+        let dealer = Dealer::new(&mut transcript, n, m);
+
+        let (parties, value_commitments): (Vec<_>, Vec<_>) = parties
+            .into_iter()
+            .enumerate()
+            .map(|(j, p)| p.assign_position(j, rng))
+            .unzip();
+
+        let (dealer, value_challenge) =
+            dealer.receive_value_commitments(&mut transcript, &value_commitments);
+
+        let (parties, poly_commitments): (Vec<_>, Vec<_>) = parties
+            .into_iter()
+            .map(|p| p.apply_challenge(&value_challenge, rng))
+            .unzip();
+
+        let (dealer, poly_challenge) =
+            dealer.receive_poly_commitments(&mut transcript, &poly_commitments);
+
+        let proof_shares: Vec<ProofShare> = parties
+            .into_iter()
+            .map(|p| p.apply_challenge(&poly_challenge))
+            .collect();
+
+        dealer.receive_shares(
+            &mut transcript,
+            &proof_shares,
+            &generators.all(),
+            value_challenge.y,
+        )
+    }
+
+    fn test_u32(m: usize) {
+        let mut rng = OsRng::new().unwrap();
+        let mut transcript = ProofTranscript::new(b"AggregatedRangeProofTest");
+
+        let v: Vec<u64> = iter::repeat(())
+            .map(|()| rng.next_u32() as u64)
+            .take(m)
+            .collect();
+        let rp = create_multi(&mut rng, v, 32);
+        assert!(rp.verify(&mut rng, &mut transcript).is_ok());
+    }
+
+    fn test_u64(m: usize) {
+        let mut rng = OsRng::new().unwrap();
+        let mut transcript = ProofTranscript::new(b"AggregatedRangeProofTest");
+
+        let v: Vec<u64> = iter::repeat(()).map(|()| rng.next_u64()).take(m).collect();
```

Comment:
Same as above: `repeat().map().taken(m)` feels a bit backwards. Although, it's purely a matter of style, so I'd defer to more experienced rustaceans.